tools/shared/src/main/scala/org/scalajs/core/tools/javascript/JSDesugaring.scala

```diff
@@ -107,59 +109,29 @@ import java.io.StringWriter
  *
  *  @author Sébastien Doeraene
  */
-object JSDesugaring {
+private[javascript] object JSDesugaring {
 
   private final val ScalaJSEnvironmentName = "ScalaJS"
 
-  /** Desugars a statement of the IR into ES5 JavaScript. */
-  @deprecated("Use an overload with an explicit OutputMode.", "0.6.2")
-  def desugarJavaScript(tree: Tree, semantics: Semantics): js.Tree =
-    desugarJavaScript(tree, semantics, OutputMode.ECMAScript51Global)
-
-  /** Desugars a statement of the IR into a given mode of JavaScript. */
-  def desugarJavaScript(tree: Tree, semantics: Semantics,
-      outputMode: OutputMode): js.Tree =
-    desugarJavaScript(tree, semantics, outputMode, Env.empty)
-
-  /** Desugars a statement of the IR into ES5 JavaScript under a
-   *  given environment.
-   */
-  @deprecated("Use an overload with an explicit OutputMode.", "0.6.2")
-  def desugarJavaScript(tree: Tree, semantics: Semantics, env: Env): js.Tree =
-    desugarJavaScript(tree, semantics, OutputMode.ECMAScript51Global, env)
-
-  /** Desugars a statement of the IR into a given mode of JavaScript under a
-   *  given environment.
-   */
-  def desugarJavaScript(tree: Tree, semantics: Semantics,
-      outputMode: OutputMode, env: Env): js.Tree = {
-    val desugar = new JSDesugar(None, semantics, outputMode)
-    try {
-      desugar.transformStat(tree)(env)
-    } catch {
-      case cause: Throwable =>
-        throw new DesugarException(tree, cause)
-    }
-  }
-
```

Comment:
Why did you remove `desugarJavaScript`? I can't see an inherent reason that we can't do it anymore (of course only given the additional info we need now).